src/main/java/org/jenkins_ci/plugins/flexible_publish/ConditionalPublisher.java

```diff
@@ -177,14 +178,32 @@ public boolean prebuild(final AbstractBuild<?, ?> build, final BuildListener lis
         return true;
     }
 
+    private static String getBuildStepName(BuildStep s) {
+        if (s instanceof Describable) {
+            return String.format("%s (%s)", ((Describable<?>)s).getDescriptor().getDisplayName(), s.toString());
+        } else {
+            return s.toString();
+        }
+    }
+
     public boolean perform(final AbstractBuild<?, ?> build, final Launcher launcher, final BuildListener listener)
                                                                                                 throws InterruptedException, IOException {
+        // do as AbstractBuild.AbstractRunner#performAllBuildSteps
+        boolean wholeResult = true;
         for (BuildStep publisher: getPublisherList()) {
-            if (!runner.perform(condition, publisher, build, launcher, listener)) {
-                return false;
```

Comment:
See, runner is not publisher, it's behaviour about "should we run next Action in ConditionalPublisher?" is chosen in configuration. If you choose "fail and don't run" it will fail build result and return false, this false is used to not continue executing any next Actions. That's why it return false for false runner. 
After your change it will execute all Actions that is wrong.